pkg/platform/yaml/yaml.go

```diff
@@ -129,33 +130,51 @@ func (ys *Yaml) Write(path, fileName string, docRead platform.KindSpecifier) err
 	return nil
 }
 
-func containsMatchingUrl(urlMethods map[string][]string, urlStr string, requestMethod models.Method) bool {
-	parsedURL, err := url.Parse(urlStr)
+func containsMatchingUrl(urlMethods []string, urlStr string, requestUrl string, requestMethod models.Method) bool {
+	urlMatched := false
+	parsedURL, err := url.Parse(requestUrl)
 	if err != nil {
 		return false
 	}
 
 	// Check for URL path and method
-	path := parsedURL.Path
-	if methods, exists := urlMethods[path]; exists {
-		// Loop through the methods for this path
-		for _, method := range methods {
-			// If the request method matches one of the allowed methods, return true
+	regex, err := regexp.Compile(urlStr)
+	if err != nil {
+		return false
+	}
+
+	urlMatch := regex.FindStringSubmatch(parsedURL.Path)
```

Comment:
Why are we using substring here..? regex will contain all the info right...? and what is that function returning anyways...?